src/providers/helpers/markUploaded.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import Realm from "realm";
+
+import realmConfig from "../../models/index";
+import { getUTCDate } from "../../sharedHelpers/dateAndTime";
+
+const markUploaded = async ( uuid: string, id: number ) => {
+  try {
+    const realm = await Realm.open( realmConfig );
+    const obs = realm.objectForPrimaryKey( "Observation", uuid );
+    realm?.write( ( ) => {
+      obs.id = id;
+      obs.timeSynced = getUTCDate( new Date( ) );
+    } );
+  } catch ( e ) {
+    console.log( e, "couldn't mark obs uploaded in realm" );
+  }
+};
+
+const markPhotoUploaded = async ( uuid: string, id: number ) => {
+  try {
+    const realm = await Realm.open( realmConfig );
+    const obsPhoto = realm.objectForPrimaryKey( "ObservationPhoto", uuid );
+    realm?.write( ( ) => {
+      obsPhoto.photo.id = id;
+      obsPhoto.photo.timeSynced = getUTCDate( new Date( ) );
+    } );
+  } catch ( e ) {
+    console.log( e, "couldn't mark photo uploaded in realm" );
+  }
+};
+
+const markSoundUploaded = async ( uuid: string, id: number ) => {
+  try {
+    const realm = await Realm.open( realmConfig );
+    const obsSound = realm.objectForPrimaryKey( "ObservationSound", uuid );
+    realm?.write( ( ) => {
+      obsSound.id = id;
+    } );
+  } catch ( e ) {
+    console.log( e, "couldn't mark photo uploaded in realm" );
+  }
+};
+
+export {
+  markUploaded,
+  markPhotoUploaded,
+  markSoundUploaded
+};
```

Comment:
All these methods are really similar. Could they be combined into something like `markRecordUploaded( uuid: string, type: string, response: object )` and put the type-specific mapping work into other functions, maybe `attrsFromReponse( response: object, type: string )`?

Reply:
Hmm, is `timeSynced` something that should be coming back from the server? And if so, is that from the `updated_at` attribute?

Reply:
No, the server doesn't know anything about when a given client synced an observation. It can only tell you when the record was updated, which might have been b/c the client updated, or another client updated it, or some server process updated it, etc.